lib/ukrand/dev.c

```diff
@@ -22,15 +22,18 @@
 static int dev_random_read(struct device *dev __unused, struct uio *uio,
 			   int flags __unused)
 {
-	size_t count;
+	__sz count;
+	__ssz ret;
 	char *buf;
 
 	buf = uio->uio_iov->iov_base;
 	count = uio->uio_iov->iov_len;
+	uio->uio_resid = 0;
 
-	uk_chacha_fill_buffer(buf, count);
+	ret = uk_random_fill_buffer(buf, count);
```

Comment:
Is it likely that in a perfectly normal working scenario this function could return an error? If not, could you plaease wrap the below `if` with an `unlikely`? Or better yet, just
```C
return uk_random_fill_buffer(buf, count);
```